From 15262dd62bee512c6c5121b43f1a1c1c78e989e4 Mon Sep 17 00:00:00 2001
From: VitalyPolezhay <>
Date: Thu, 21 Sep 2023 21:27:46 +0300
Subject: [PATCH] fixed bug with 2d initial image view

---
 .../SkinCreatorViewController.swift                    | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/Crafty Craft 5/Tab Bar/Create Tab/SkinsCreator/ListOfCreatedSkinsController/Skin2DCreator/BodyPart2DPickerVC/SkinCreatorController/SkinCreatorViewController.swift b/Crafty Craft 5/Tab Bar/Create Tab/SkinsCreator/ListOfCreatedSkinsController/Skin2DCreator/BodyPart2DPickerVC/SkinCreatorController/SkinCreatorViewController.swift
index 3762bf1..992dfe0 100755
--- a/Crafty Craft 5/Tab Bar/Create Tab/SkinsCreator/ListOfCreatedSkinsController/Skin2DCreator/BodyPart2DPickerVC/SkinCreatorController/SkinCreatorViewController.swift	
+++ b/Crafty Craft 5/Tab Bar/Create Tab/SkinsCreator/ListOfCreatedSkinsController/Skin2DCreator/BodyPart2DPickerVC/SkinCreatorController/SkinCreatorViewController.swift	
@@ -359,10 +359,12 @@ class SkinCreatorViewController: UIViewController, PKCanvasViewDelegate, PKToolP
         
         var colorsArr: [UIColor]?
         
-        colorsArr = currentEditableSkin?.skinAssemblyDiagram?.extractPixelColors(width: currentBodyPartSide?.width,
-                                                                                 height: currentBodyPartSide?.height,
-                                                                                 startX: currentBodyPartSide?.startX,
-                                                                                 startY: currentBodyPartSide?.startY )
+        colorsArr = currentEditableSkin?.skinAssemblyDiagram?
+            .rotate(radians: CGFloat.pi/2)?
+            .extractPixelColors(width: currentBodyPartSide?.width,
+                                height: currentBodyPartSide?.height,
+                                startX: currentBodyPartSide?.startX,
+                                startY: currentBodyPartSide?.startY)
         
         let sceneSize = CGSize(width: SCREEN_WIDTH, height: SCREEN_HEIGHT)
         let canvasSize = CGSize(width: CANVAS_WIDTH, height: CANVAS_HEIGHT)
-- 
2.37.0

